docs/API.md

```diff
@@ -129,21 +130,22 @@ Returns a `Client` instance and perform login.
- * connect : a function taking the client as parameter and that should client.setSocket(socket) 
+ * connect : a function taking the client as parameter and that should client.setSocket(socket)
  and client.emit('connect') when appropriate (see the proxy examples for an example of use)
- * agent : a http agent that can be used to set proxy settings for yggdrasil authentication (see proxy-agent on npm) 
+ * agent : a http agent that can be used to set proxy settings for yggdrasil authentication (see proxy-agent on npm)
  * fakeHost : (optional) hostname to send to the server in the set_protocol packet
  * profilesFolder : optional
-   * (mojang account) the path to the folder that contains your `launcher_profiles.json`. defaults to your minecraft folder if it exists, otherwise the local directory. set to `false` to disable managing profiles 
+   * (mojang account) the path to the folder that contains your `launcher_profiles.json`. defaults to your minecraft folder if it exists, otherwise the local directory. set to `false` to disable managing profiles
    * (microsoft account) the path to store authentication caches, defaults to .minecraft
  * onMsaCode(data) : (optional) callback called when signing in with a microsoft account
  with device code auth. `data` is an object documented [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code#device-authorization-response)
  * id : a numeric client id used for referring to multiple clients in a server
  * validateChannelProtocol (optional) : whether or not to enable protocol validation for custom protocols using plugin channels. Defaults to true
  * disableChatSigning (optional) : Don't try obtaining chat signing keys from Mojang (1.19+)
- * realms : An object which should contain one of the following properties: `realmId` or `pickRealm`. When defined will attempt to join a Realm without needing to specify host/port. **The authenticated account must either own the Realm or have been invited to it** 
-   * realmId : The id of the Realm to join. 
-   * pickRealm(realms) : A function which will have an array of the user Realms (joined/owned) passed to it. The function should return a Realm. 
+ * realms : An object which should contain one of the following properties: `realmId` or `pickRealm`. When defined will attempt to join a Realm without needing to specify host/port. **The authenticated account must either own the Realm or have been invited to it**
+   * realmId : The id of the Realm to join.
+   * pickRealm(realms) : A function which will have an array of the user Realms (joined/owned) passed to it. The function should return a Realm.
+ * Client : You can pass a custom client class to use instead of the default one, which would allow you to create completely custom communication. Also note that you can use the `stream` option instead where you can supply custom duplex, but this will still use serialization/deserialization of packets.
```

Comment:
it would be good to say here what methods this class must expose

Reply:
I guess there is the typing for that (with all signatures etc)